streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -259,7 +259,7 @@ private void setState(final State newState) {
                 log.info("{} State transition from {} to {}.", logPrefix, oldState, newState);
             }
             state = newState;
-            if (stateListener != null) {
+            if (stateListener != null && state != oldState) {
```

Comment:
\cc @enothereska Just want to bring this to your attention as you recently worked on this.